### PR TITLE
testing/waybar: new aport

### PR DESCRIPTION
**Pull request by z3ntu**

### CODE REVIEW

File: testing/fmt/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Contributor:
+# Maintainer:
+pkgname="fmt"
+pkgver="5.3.0"
+pkgrel=0
+pkgdesc="Open-source formatting library for C++"
+url="https://fmt.dev/latest/index.html"
+arch="all"
+license="BSD-2-Clause"
+depends=""
```
Comment (maxice8) [inline testing/fmt/APKBUILD]: empty variables can be removed for brevity

File: testing/fmt/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+depends_dev=""
+makedepends="$depends_dev"
```
Comment (maxice8) [inline testing/fmt/APKBUILD]: makedepends can be removed

File: testing/fmt/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Contributor:
+# Maintainer:
```
Comment (maxice8) [inline testing/fmt/APKBUILD]: Aren't you willing to maintain them ?

File: testing/spdlog/APKBUILD [hunk 4]
```diff
@@ -8,7 +8,7 @@ options="!check" # Unable to run tests
 url="https://github.com/gabime/spdlog"
 arch="all"
 license="MIT"
-makedepends="cmake"
+makedepends="cmake fmt-dev"
```
Comment (maxice8) [inline testing/spdlog/APKBUILD]: These changes require spdlog to have its pkgrel bumped

File: testing/waybar/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Contributor:
+# Maintainer:
+pkgname="waybar"
+pkgver="0.7.1"
+pkgrel=0
+pkgdesc="Highly customizable Wayland bar for Sway and Wlroots based compositors"
+url="https://github.com/Alexays/Waybar/"
+arch="all"
+license="MIT"
+depends=""
```
Comment (maxice8) [inline testing/waybar/APKBUILD]: empty variables can be removed

File: testing/waybar/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Contributor:
+# Maintainer:
```
Comment (maxice8) [inline testing/waybar/APKBUILD]: Same question about maintainership for fmt applies here

File: testing/waybar/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+		--mandir=/usr/share/man \
+		--localstatedir=/var \
+		--buildtype=release \
+		. output
+	ninja -C output
+}
+
+check() {
+	# Replace with proper check command(s)
+	:
```
Comment (maxice8) [inline testing/waybar/APKBUILD]: This needs to be fixed or a !check with a small comment telling why there is no check enabled made.

File: testing/waybar/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Contributor: Luca Weiss <luca@z3ntu.xyz>
+# Maintainer: Luca Weiss <luca@z3ntu.xyz>
+pkgname="waybar"
+pkgver="0.7.1"
+pkgrel=0
+pkgdesc="Highly customizable Wayland bar for Sway and Wlroots based compositors"
+url="https://github.com/Alexays/Waybar/"
+arch="all"
+license="MIT"
+makedepends="meson gtkmm3-dev jsoncpp-dev spdlog-dev fmt-dev libinput-dev libnl-dev pulseaudio-dev libmpdclient-dev libgudev-dev"
```
Comment (maxice8) [inline testing/waybar/APKBUILD]: Can this line be cut at about 80 chars or less ?

Comment (maxice8) [reply testing/waybar/APKBUILD]: not outdated

**Comment by z3ntu** [comment]
I'm actually not sure if the spdlog change worked because spdlog still installs the bundled fmt headers...

**Comment by maxice8** [comment]
> I'm actually not sure if the spdlog change worked because spdlog still installs the bundled fmt headers...

Can't they just be deleted ?

**Comment by maxice8** [comment]
`fmt` install the pkgconfig file to /usr/share, please make it install to /usr/lib

**Comment by z3ntu** [comment]
Fixed. I still have to test this on a device and figure out what difference the spdlog commit does; but I think that's all for this PR. Thanks for the extremely quick reviews @maxice8 :)

**Comment by z3ntu** [comment]
So the spdlog commit makes this difference in the built `spdlog-dev` package (the `spdlog` package stays the same):
```
│ ├── usr/lib/cmake/spdlog/spdlogTargets.cmake
│ │ @@ -59,15 +59,17 @@
│ │    set(_IMPORT_PREFIX "")
│ │  endif()
│ │  
│ │  # Create imported target spdlog::spdlog
│ │  add_library(spdlog::spdlog INTERFACE IMPORTED)
│ │  
│ │  set_target_properties(spdlog::spdlog PROPERTIES
│ │ +  INTERFACE_COMPILE_DEFINITIONS "SPDLOG_FMT_EXTERNAL"
│ │    INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
│ │ +  INTERFACE_LINK_LIBRARIES "fmt::fmt"
│ │  )
│ │  
│ │  if(CMAKE_VERSION VERSION_LESS 3.0.0)
│ │    message(FATAL_ERROR "This file relies on consumers using CMake 3.0.0 or greater.")
│ │  endif()
│ │  
│ │  # Load information for each installed configuration.
│ ├── usr/lib/cmake/spdlog/spdlogConfig.cmake
│ │ @@ -17,15 +17,15 @@
│ │  # * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
│ │  # * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
│ │  # * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
│ │  # * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
│ │  # * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
│ │  # *************************************************************************/
│ │  
│ │ -set(SPDLOG_FMT_EXTERNAL OFF)
│ │ +set(SPDLOG_FMT_EXTERNAL ON)
│ │  
│ │  include("${CMAKE_CURRENT_LIST_DIR}/spdlogTargets.cmake")
│ │  
│ │  if(SPDLOG_FMT_EXTERNAL)
│ │      include(CMakeFindDependencyMacro)
│ │      find_dependency(fmt CONFIG)
│ │  endif()
│ ├── usr/lib/cmake/spdlog/spdlogConfigVersion.cmake
│ │ @@ -36,15 +36,15 @@
│ │  # if the installed project requested no architecture check, don't perform the check
│ │  if("FALSE")
│ │    return()
│ │  endif()
│ │  
│ │  # if the installed or the using project don't have CMAKE_SIZEOF_VOID_P set, ignore it:
│ │  if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "" OR "8" STREQUAL "")
│ │ -   return()
│ │ +  return()
│ │  endif()
│ │  
│ │  # check that the installed version has the same 32/64bit-ness as the one which is currently searching:
│ │  if(NOT CMAKE_SIZEOF_VOID_P STREQUAL "8")
│ │    math(EXPR installedBits "8 * 8")
│ │    set(PACKAGE_VERSION "${PACKAGE_VERSION} (${installedBits}bit)")
│ │    set(PACKAGE_VERSION_UNSUITABLE TRUE)
```

**Comment by maxice8** [comment]
So i think  the bundled stuff can be removed from include/

anyways ping me here or on IRC and i'll merge this

**Comment by z3ntu** [comment]
@maxice8 I've removed the spdlog patch for now as I've written the Arch Linux spdlog maintainer why he removed the patch which was deleting the bundled code from spdlog. I'll open a PR later if it makes any sense.

And I've added `ttf-font-awesome` as dependency for waybar as it uses that font for all icons in the bar.

This should be good to go now!

**Pull request closed without merging**